### PR TITLE
Add HiDPI scaling for decorations

### LINKED ISSUE #90: Left bottom corner pixel in CSD has a different color
picture

### LINKED ISSUE #91: Hi-DPI aware decorations
The `ConceptFrame` currently provided with SCTK is not dpi-aware and could easily be.

### PR DESCRIPTION
**Pull request by kchibisov**
Fixes #91.
Fixes #90.

### CODE REVIEW

File: src/window/concept_frame.rs [hunk 1]
```diff
@@ -112,12 +112,33 @@ impl Part {
...
+            crate::surface::setup_surface(
+                compositor.create_surface(),
+                Some(
+                    move |dpi, surface: wl_surface::WlSurface, _ddata: DispatchData| {
+                        surface.set_buffer_scale(dpi);
+                        surface.commit();
+                    },
+                ),
+            )
+        };
```
Comment (kchibisov) [inline src/window/concept_frame.rs]: I'm not sure if this is something desirable tbh. I'm not an expert in plumbing and sctk internals, so suggestions are welcome.

**Comment by vberger** [comment]
I'll keep this waiting until we get to the bottom of #102, given the issue was introduced by #95

**Comment by kchibisov** [comment]
@vberger the issue wasn't introduced by #95 it became more visible due to it. Previously the parallel line was drawn in a bit odd way due to anti-aliasing. In fact on #90 this black pixel is a separator line. I can send you a picture from old SCTK before #95, you'll see the same blue lines.

File: src/window/concept_frame.rs (hunk 1 above)
Comment (vberger) [reply src/window/concept_frame.rs]: I'm alright with that.

**Comment by vberger** [comment]
Okay, so this will need to be rebased, and then I'm good with merging it.

**Pull request merged**